### PR TITLE
Adding Jenkins file

### PR DESCRIPTION
**Pull request by frznlogic**

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/groovy
+def workspace = "/mnt/ssd/jenkins/workspace/SoftwareContainerPipeline"
```
Comment (DunderRoffe) [inline Jenkinsfile]: Seems awfully specific. Will this run on any of the nodes ?

Comment (tobsan) [inline Jenkinsfile]: Can we really have this hard-coded? We should be able to capture the workspace of the first workspace (build stage) and then use that in runInVagrant. I don't know about groovy scripts, but in Jenkins shell scripts you can use ${WORKSPACE}

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        "
+}
+
+node {
+    stage 'Download'
+    git branch: 'master', url: 'https://github.com/Pelagicore/softwarecontainer.git'
+    sh 'git submodule update --init'
+    
+    stage 'Build'
+    sh 'cd /mnt/ssd/jenkins/workspace/SoftwareContainerPipeline && vagrant up --provision'
```
Comment (tobsan) [inline Jenkinsfile]: Also, if we want the workspace variable, it should be used here as well.

**Review by tobsan — CHANGES_REQUESTED** [review]

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+// TBD: Haven't figured out how to make the parallel jobs run on the same slave/agent 
+// or if it is possible. Very annoying. Let's run sequentially in a single slave for now.
+//    parallel ('Documentation': {
+        stage 'Documentation'
+        runInVagrant('cd softwarecontainer/build && make doc')
+        
+//    }, 'UnitTest':{
+        stage 'UnitTest' 
+        runInVagrant("cd softwarecontainer/build && sudo ./run-tests.sh")
```
Comment (DunderRoffe) [inline Jenkinsfile]: Are there not problems with running the tests in sudo ?

**Review by DunderRoffe — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 1 above)
Comment (DunderRoffe) [reply Jenkinsfile]: The solution @tobsan suggested sounds good

Comment (frznlogic) [reply Jenkinsfile]: There is no $WORKSPACE in pipeline scripts unfortunately..... :/ I did find a workaround however. pwd()....

File: Jenkinsfile (hunk 2 above)
Comment (frznlogic) [reply Jenkinsfile]: Yepp, that was the idea ;)

File: Jenkinsfile (hunk 1 above)
Comment (frznlogic) [reply Jenkinsfile]: Nope :)

File: Jenkinsfile (hunk 3 above)
Comment (frznlogic) [reply Jenkinsfile]: aaah, that explains why the pulseaudio test fails perhaps :)

**Review by DunderRoffe — APPROVED** [review]

**Pull request merged**